platforms/documentation/docs/src/docs/release/notes.md

```diff
@@ -43,90 +65,182 @@ For Java, Groovy, Kotlin, and Android compatibility, see the [full compatibility
+This reduces build times and improves developer productivity.
 
-The biggest advantage of the new mechanism is the ability to work in the presence of inline functions, something that Gradle wasn't handling efficiently until now.
-This results in substantial performance gains depending on your build and the changes made to the build logic.
+The improvement comes from significantly better detection of ABI (Application Binary Interface) changes, made possible by using Kotlin’s built-in ABI fingerprinting instead of Gradle’s previous internal mechanism.
+This brings major performance benefits, especially in builds that use inline functions, which were not handled efficiently before.
 
-For example, in the `gradle/gradle` project, non-ABI changes to build logic now result in up to a 60% reduction in configuration time by avoiding unnecessary script recompilation.
+For example, in the [Gradle build](https://github.com/gradle/gradle) itself, non-ABI changes to build logic result in up to a 60% reduction in configuration time by avoiding unnecessary script recompilation.
 
 ![Reduction in unnecessary script recompilation](release-notes-assets/help_after_nonABI_change_in_buildSrc.png)
 
-#### Gradle API now uses JSpecify Nullability Annotations
+<a name="jspecify"></a>
+#### Gradle API uses JSpecify Nullability annotations
+
+Since Gradle 5.0 we've been using annotations from [JSR-305](https://jcp.org/en/jsr/detail?id=305) to make the nullness of type usages explicit for the Gradle API.
+Starting with Gradle 9, the Gradle API is annotated using [JSpecify](https://jspecify.dev/) instead.
+
+Kotlin 2.1, when combined with JSpecify annotations in the Gradle API, introduces stricter nullability handling.
+For more details and potential breakages, see the dedicated [upgrading guide section](userguide/upgrading_version_8.html#jspecify).
+
+<a name="sem-ver-wrapper"></a>
+#### Support for major and minor version specification in Gradle Wrapper
+
+Gradle supports specifying only a major or minor version when configuring the [wrapper](userguide/gradle_wrapper.html).
+
+For example, the following resolves to the latest `9.x.y` release:
+
+```
+./gradlew wrapper --gradle-version=9
+```
+
+While the following resolves to the latest `9.1.x` release:
+
+```
+./gradlew wrapper --gradle-version=9.1
+```
+
+This feature requires Gradle 9.0.0 or later.
+Earlier versions don’t follow full semantic versioning and may misinterpret partial versions (e.g., `8.12` might refer to `8.12` (because it's an exact version) and `8.12.1` (semantically the latest version for `8.12`).
+
+Gradle’s [version information endpoint](https://services.gradle.org/versions/) has been extended to support this behavior. For instance, [https://services.gradle.org/versions/9](https://services.gradle.org/versions/9) lists all versions of Gradle with major version 9.
+
+#### Archive tasks produce reproducible archives by default
```

Comment:
This should probably be a release highlight

Reply:
To discuss for RC2/GA